### PR TITLE
feat(github-workflow): Add build and publish pipeline

### PR DESCRIPTION
**Pull request by alessio-creo**
No changes in application code

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  IMAGE_NAME: ${{ github.repository }}
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!main'
+      - '!master'
+    paths:
+      - '*'
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
      - '**'
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]: Where is the `buildx` step? 🤔

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM maven:3.8.3-openjdk-17 AS MAVEN_BUILD
+
+WORKDIR /interop-be-probing-api/
+
+COPY . .
+
+RUN mvn clean package -Dmaven.test.skip=true
```
Comment (micdes-pagopa) [inline Dockerfile]: Can we stop maven from logging the download operations? The GH action log gets hard to read 😵‍💫 https://github.com/pagopa/interop-be-probing-api/actions/runs/4356188202/jobs/7613846107#step:5:209 
I don't know if it's possibile, but it's worth trying

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!main'
+      - '!master'
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml:8]: Are you going to create a different workflow for `main`? I think we can remove these filters and use this workflow

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        - warning
+        - debug
+      environment:
+        description: 'Which environment to update.'
+        type: choice
+        required: true
+        default: dev
+        options:
+        - dev
+        - uat
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]: Do you see a scenario where we need to manually launch a build without new commits?

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+jobs:
+  build-and-push-image:
+    if: ${{ github.event_name == 'push' }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]: This `if` is not necessary if the workflow runs on every push (line 9)

File: .github/workflows/build.yaml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
          images: ghcr.io/${{ github.repository }}
```
Just my preference, it's easier to double check 😄

File: .github/workflows/build.yaml [hunk 8]
```diff
@@ -0,0 +1,69 @@
+name: "CI"
+
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
```
See suggestion at line 60

File: .github/workflows/build.yaml (hunk 2 above)
Comment (alessio-creo) [reply .github/workflows/build.yaml]: Probabily in the prebuilt action, I'll investigate

File: Dockerfile (hunk 3 above)
Comment (alessio-creo) [reply Dockerfile]: I know. I'll investigate on mvn options

File: .github/workflows/build.yaml (hunk 4 above)
Comment (alessio-creo) [reply .github/workflows/build.yaml:8]: We will probably differentiate the workflows but not at the moment. I'll change that

File: .github/workflows/build.yaml (hunk 5 above)
Comment (alessio-creo) [reply .github/workflows/build.yaml]: Maybe for not code related errors. If doesn't make sense for you w e can remove this.

File: .github/workflows/build.yaml (hunk 6 above)
Comment (alessio-creo) [reply .github/workflows/build.yaml]: Agreed

File: .github/workflows/build.yaml (hunk 7 above)
Comment (alessio-creo) [reply .github/workflows/build.yaml]: Do you have any hint to use the image sha in the name? Alternatevely we can use the commit sha

File: .github/workflows/build.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        - warning
+        - debug
+      environment:
+        description: 'Which environment to update.'
+        type: choice
+        required: true
+        default: dev
+        options:
+        - dev
+        - uat
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
```

File: .github/workflows/build.yaml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
          registry: ghcr.io
```

File: .github/workflows/build.yaml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        type: choice
+        required: true
+        default: dev
+        options:
+        - dev
+        - uat
+
+jobs:
+  build-and-push-image:
+    if: ${{ github.event_name == 'push' }}
```
Comment (micdes-pagopa) [inline .github/workflows/build.yaml]:
```suggestion
```

**Pull request merged**